src/lib/configs/project/dockerCompose.ts

```diff
@@ -329,3 +401,162 @@ export const initReadonlyDockerComposeConfig =
   getReadonlyConfigInitFunction(initConfigOptions);
 
 export const dockerComposeSchema: JSONSchemaType<LatestConfig> = configSchemaV0;
+
+export async function ensureConfigToml(
+  fluenceConfig: FluenceConfigReadonly | null,
+  providerConfig: ProviderConfigReadonly,
+) {
+  const baseNoxConfig = mergeNoxConfigYAML(
+    await getDefaultNoxConfigYAML(fluenceConfig),
+    providerConfig.nox ?? {},
+  );
+
+  const configsDir = await ensureFluenceConfigsDir();
+  const { stringify } = await import("@iarna/toml");
+
+  await Promise.all(
+    Object.entries(providerConfig.computePeers).map(async ([key, value], i) => {
+      const overridden = mergeNoxConfigYAML(baseNoxConfig, value.nox ?? {});
+
+      if (overridden.tcpPort === TCP_PORT_START) {
+        overridden.tcpPort = TCP_PORT_START + i;
+      }
+
+      if (overridden.websocketPort === WEB_SOCKET_PORT_START) {
+        overridden.websocketPort = WEB_SOCKET_PORT_START + i;
+      }
+
+      if (overridden.httpPort === HTTP_PORT_START) {
+        overridden.httpPort = HTTP_PORT_START + i;
+      }
+
+      return writeFile(
+        join(configsDir, getConfigTomlName(key)),
+        [
+          stringify(configYAMLToConfigToml(overridden)),
+          providerConfig.nox?.rawConfig ?? "",
+          value.nox?.rawConfig ?? "",
+        ]
+          .filter((x) => {
+            return x !== "";
+          })
+          .join("\n"),
+        FS_OPTIONS,
+      );
+    }),
+  );
+}
+
+function mergeNoxConfigYAML(a: NoxConfigYAML, b: NoxConfigYAML): NoxConfigYAML {
+  return {
+    ...a,
+    ...b,
+    ...(a.systemServices !== undefined && b.systemServices !== undefined
+      ? {
+          systemServices: {
+            ...a.systemServices,
+            ...b.systemServices,
+            enable: Array.from(
+              new Set([
+                ...(a.systemServices.enable ?? []),
+                ...(b.systemServices.enable ?? []),
+              ]),
+            ),
+            aquaIpfs: {
+              ...a.systemServices.aquaIpfs,
+              ...b.systemServices.aquaIpfs,
+            },
+            decider: {
+              ...a.systemServices.decider,
+              ...b.systemServices.decider,
+            },
+          },
+        }
+      : {}),
+  };
+}
+
+function configYAMLToConfigToml(config: NoxConfigYAML) {
+  const configToml: NoxConfigTOML = {};
+
+  if (config.aquavmPoolSize !== undefined) {
+    configToml.aquavm_pool_size = config.aquavmPoolSize;
```

Comment:
OMG 2 
Can we use something like `https://www.npmjs.com/package/change-case`

Reply:
there is the function to change case in lodash but I didn't use it because 
first it wasn't too big when I started
second I wanted to retain some type information
third I'v been thinking that not everything will be as straight forward as `camelCase` -> `snake_case`

but seems it got a bit out of hand, you are right, It will be less error-prone if we don't care about types and just transform this to snake_case. If there will be some differenences in the future - then I think I will extend this function so it does some custom transaformations when and if they will be required in the future